scRNA-seq-advanced/02-celltype_assignment.Rmd

```diff
@@ -44,149 +49,233 @@ set.seed(12345)
-Here we are using the `tidyr::pivot_longer()` function, which allows us to convert our data frame with one column per tag into a data frame with separate columns for the tag id (`ADT`) and the expression value (`logcount`).
+If we just wanted to plot one of these tags, we could do so right away, but with a bit more data wrangling, we can convert these results into a "tidier" format, that will allow us to take full advantage of `tidyverse` tools!
 
-```{r}
+Right now the data is in a "wide" format, with each column a different tag. 
+But the data in all of the columns is the same type, and measures something similar: the normalized count of an ADT.
+One could even argue that each row contains 10 different observations, where the "tidy" data ideal, as espoused by [Wickham (2014)](https://doi.org/10.18637/jss.v059.i10), requires a single observation per row, a "long" format.
+This long format will have one column that tells us which ADT was measured and a second column with the measurement value itself.
+
+We can perform this conversion using the `tidyr::pivot_longer()` function.
+Following conversion, we can filter to just the ADTs that we care about.
+which allows us to convert our data frame with one column per tag into a data frame with separate columns for the tag id (`ADT`) and the expression value (`logcount`).
+
+```{r pivot longer, live=TRUE}
 adt_df_long <- adt_df %>%
-  dplyr::select(c("CD3", "CD4", "CD8")) %>%
-  tidyr::pivot_longer(everything(), names_to = "ADT", values_to = "logcount")
-```
+  # pivot to long format
+  tidyr::pivot_longer(
+    everything(), # use all columns
+    names_to = "ADT", # convert row names to a column called "ADT"
+    values_to = "logcount" # name the value column "logcount"
+  ) %>%
+  # filter to tags we are interested in
+  dplyr::filter(ADT %in% c("CD3", "CD4", "CD8"))
 
+# look at the resulting df
+head(adt_df_long)
+```
 
-Now we can make a density plot with `ggplot2`!
+Now we can make a density plot with `ggplot2` for all three ADTs we are interested in at once.
 
-```{r}
+```{r plot ADTs, live=TRUE}
+# plot logcounts by ADT
 ggplot(adt_df_long, aes(x = logcount, fill = ADT)) + 
-  geom_density() +
-  scale_x_continuous(breaks = seq(0, 10, by = 2)) + 
-  facet_wrap(~ADT, ncol = 1) +
-  theme_bw() +
-  theme(legend.position = "none")
+  geom_density() + # density plot
+  facet_wrap(~ADT, ncol = 1) + # facet by ADT
+  theme_bw() + # nicer theme
+  theme(legend.position = "none") # no legend needed
 ```
 
-Looking at a few of these, we can define some rough cutoffs.
-This is very manual and not particularly principled, but it _is_ a potential way to assign cell types.
+These look pretty good!
+There seem to be different baseline levels for each of the different tags, but each has a pretty nice bimodal distribution.
 
-```{r}
+We can now use the values from these plots to construct a set of rules to classify the T cells. 
+We will do this using the "wide" data frame from earlier. 
```

Comment:
Someone will ask "why/when use wide vs long" and the answer will just have to be "depends on what you are doing and which format makes more sense for your task."